public/app/features/serviceaccounts/components/CreateTokenModal.tsx

```diff
@@ -24,34 +25,56 @@ const EXPIRATION_OPTIONS = [
 
 export type ServiceAccountToken = {
   name: string;
-  secondsToLive: number;
+  secondsToLive?: number;
 };
 
-interface CreateTokenModalProps {
+interface Props {
   isOpen: boolean;
   token: string;
+  serviceAccountLogin: string;
   onCreateToken: (token: ServiceAccountToken) => void;
   onClose: () => void;
 }
 
-export const CreateTokenModal = ({ isOpen, token, onCreateToken, onClose }: CreateTokenModalProps) => {
+export const CreateTokenModal = ({ isOpen, token, serviceAccountLogin, onCreateToken, onClose }: Props) => {
+  let tomorrow = new Date();
+  tomorrow.setDate(tomorrow.getDate() + 1);
+
+  const [defaultTokenName, setDefaultTokenName] = useState('');
   const [newTokenName, setNewTokenName] = useState('');
   const [isWithExpirationDate, setIsWithExpirationDate] = useState(false);
-  const [newTokenExpirationDate, setNewTokenExpirationDate] = useState<Date | string>('');
-  const [isExpirationDateValid, setIsExpirationDateValid] = useState(false);
+  const [newTokenExpirationDate, setNewTokenExpirationDate] = useState<Date | string>(tomorrow);
+  const [isExpirationDateValid, setIsExpirationDateValid] = useState(newTokenExpirationDate !== '');
   const styles = useStyles2(getStyles);
 
+  useEffect(() => {
+    // Generate new token name every time we open modal
+    if (isOpen) {
+      setDefaultTokenName(`${serviceAccountLogin}-${uuidv4()}`);
+    }
+    // Cleanup on unmount
+    return () => setDefaultTokenName('');
```

Comment:
Not sure cleanup is necessary for the local state, it will be reset in any case if component does unmount.

Reply:
Yeah, true!